application/client.core/src/app/environment/components/sidebar/search.manager/charts/list/component.ts

```diff
@@ -57,6 +62,18 @@ export class SidebarAppSearchManagerChartsComponent implements OnDestroy, AfterC
         this.provider.select().doubleclick(event, entity);
     }
 
+    public _ng_onDragStarted(entity: Entity<ChartRequest>) {
+        this._dragging = entity;
+        SearchManagerService.onDragStart(true);
+    }
+
+    private _onRemove() {
+        if (this._dragging) {
+            this._dragging.getEntity().remove(this.provider.getSession());
+        }
+        this._dragging = undefined;
+    }
```

Comment:
This code is repeated for filters, charts, time ranges, and disabled entities. 
Let's try to move out this code into `directive`. It will allow inherit it and reuse for each entity.